hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpScheduledTransactionsSubcommand.java

```diff
@@ -0,0 +1,252 @@
+            @NonNull final DumpStateCommand.EmitSummary emitSummary,
+            @NonNull final SignedStateCommand.Verbosity verbosity) {
+        new DumpScheduledTransactionsSubcommand(state, scheduledTxsPath, emitSummary, verbosity).doit();
+    }
+
+    @NonNull
+    final SignedStateHolder state;
+
+    @NonNull
+    final Path scheduledTxsPath;
+
+    @NonNull
+    final DumpStateCommand.EmitSummary emitSummary;
+
+    @NonNull
+    final SignedStateCommand.Verbosity verbosity;
+
+    DumpScheduledTransactionsSubcommand(
+            @NonNull final SignedStateHolder state,
+            @NonNull final Path scheduledTxsPath,
+            @NonNull final DumpStateCommand.EmitSummary emitSummary,
+            @NonNull final SignedStateCommand.Verbosity verbosity) {
+        requireNonNull(state, "state");
+        requireNonNull(scheduledTxsPath, "signedTxsPath");
+        requireNonNull(emitSummary, "emitSummary");
+        requireNonNull(verbosity, "verbosity");
+
+        this.state = state;
+        this.scheduledTxsPath = scheduledTxsPath;
+        this.emitSummary = emitSummary;
+        this.verbosity = verbosity;
+    }
+
+    void doit() {
+        final var scheduledTransactions = state.getScheduledTransactions().byId();
+        System.out.printf("=== %d scheduled transactions ===%n", scheduledTransactions.size());
+
+        final var allScheduledTxs = gatherScheduledTransactions(scheduledTransactions);
+
+        int reportSize;
+        try (@NonNull final var writer = new Writer(scheduledTxsPath)) {
+            if (emitSummary == DumpStateCommand.EmitSummary.YES) reportSummary(writer, allScheduledTxs);
+            reportOnScheduledTransactions(writer, allScheduledTxs);
+            reportSize = writer.getSize();
+        }
+
+        System.out.printf("=== scheduled transactions report is %d bytes%n", reportSize);
+    }
+
+    @SuppressWarnings(
+            "java:S6218") // "Equals/hashcode method should be overridden in records containing array fields" - this
+    record ScheduledTransaction(
+            long number,
+            @NonNull Key grpcAdminKey,
+            @NonNull Optional<JKey> adminKey,
+            @NonNull String memo,
+            boolean deleted,
+            boolean executed,
+            boolean calculatedWaitForExpiry,
+            boolean waitForExpiryProvided,
+            @Nullable EntityId payer,
+            EntityId schedulingAccount,
+            RichInstant schedulingTXValidStart,
+            @Nullable RichInstant expirationTimeProvided,
+            @Nullable RichInstant calculatedExpirationTime,
+            @Nullable RichInstant resolutionTime,
+            byte[] bodyBytes,
+            TransactionBody ordinaryScheduledTxn,
+            SchedulableTransactionBody scheduledTxn,
+            List<byte[]> signatories) {
+
+        ScheduledTransaction(@NonNull final ScheduleVirtualValue scheduledTx) {
+            this(
+                    scheduledTx.getKey().getKeyAsLong(),
```

Comment:
perhaps a `requiresNonNull` around the first `scheduledTx` here?  (This will be sufficient as Java evaluates the arguments left-to-right.)

Reply:
actually I see that though you're pretty good about `@NonNull` annotations you haven't used `requireNonNull` much down in the class's internal methods.  I'm not that strict about it here; add them or not as you wish down in those other methods, but here, in the constructor, there probably should be one.